File: specification/devcenter/resource-manager/Microsoft.DevCenter/preview/2022-08-01-preview/examples-java/DevBoxDefinitions_ListByProject.java
import com.azure.core.util.Context;

/** Samples for DevBoxDefinitions ListByProject. */
public final class Main {
    /*
     * x-ms-original-file: specification/devcenter/resource-manager/Microsoft.DevCenter/preview/2022-08-01-preview/examples/DevBoxDefinitions_ListByProject.json
     */
    /**
     * Sample code: DevBoxDefinitions_ListByProject.
     *
     * @param manager Entry point to DevCenterManager.
     */
    public static void devBoxDefinitionsListByProject(com.azure.resourcemanager.devcenter.DevCenterManager manager) {
        manager.devBoxDefinitions().listByProject("rg1", "ContosoProject", null, Context.NONE);
    }
}

File: specification/devcenter/resource-manager/Microsoft.DevCenter/preview/2022-09-01-preview/examples-js/DevBoxDefinitions_ListByProject.js
const { DevCenterClient } = require("@azure/arm-devcenter");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to List Dev Box definitions configured for a project.
 *
 * @summary List Dev Box definitions configured for a project.
 * x-ms-original-file: specification/devcenter/resource-manager/Microsoft.DevCenter/preview/2022-09-01-preview/examples/DevBoxDefinitions_ListByProject.json
 */
async function devBoxDefinitionsListByProject() {
  const subscriptionId = "{subscriptionId}";
  const resourceGroupName = "rg1";
  const projectName = "ContosoProject";
  const credential = new DefaultAzureCredential();
  const client = new DevCenterClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.devBoxDefinitions.listByProject(resourceGroupName, projectName)) {
    resArray.push(item);
  }
  console.log(resArray);
}

devBoxDefinitionsListByProject().catch(console.error);
